src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -0,0 +1,546 @@
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(Zcl, "HandleProvideOffer: Invalid stream usage");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ResourceExhausted);
+            return;
+        }
+
+        // Prepare delegate arguments.
+        Delegate::ProvideOfferRequestArgs args;
+        args.sessionId       = GenerateSessionId();
+        args.streamUsage     = req.streamUsage;
+        args.videoStreamId   = videoStreamID;
+        args.audioStreamId   = audioStreamID;
+        args.metadataOptions = req.metadataOptions;
+        args.peerNodeId      = peerNodeId;
+        args.sdp             = std::string(req.sdp.data(), req.sdp.size());
+
+        // Convert ICE servers list from DecodableList to vector.
+        if (req.ICEServers.HasValue())
+        {
+            auto iterator = req.ICEServers.Value().begin();
+            while (iterator.Next())
+            {
+                args.iceServers.Value().push_back(iterator.GetValue());
+            }
+        }
+
+        // Convert ICETransportPolicy from CharSpan to std::string.
+        if (req.ICETransportPolicy.HasValue())
+        {
+            args.iceTransportPolicy.SetValue(
+                std::string(req.ICETransportPolicy.Value().data(), req.ICETransportPolicy.Value().size()));
+        }
+
+        // Delegate processing: process the SDP offer, gather ICE candidates, create SDP answer, etc.
+        auto delegateStatus = Protocols::InteractionModel::ClusterStatusCode(mDelegate.HandleProvideOffer(args, outSession));
+        if (!delegateStatus.IsSuccess())
+        {
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, delegateStatus);
+            return;
+        }
+
+        // Store the new WebRTCSessionStruct in CurrentSessions.
+        UpsertSession(outSession);
+    }
+    else
+    {
+        // Re-offer: webRTCSessionID is not null.
+        uint16_t sessionId                    = webRTCSessionID.Value();
+        WebRTCSessionStruct * existingSession = FindSession(sessionId);
+
+        // If the session does not exist, respond with NOT_FOUND.
+        if (!existingSession)
+        {
+            ChipLogError(Zcl, "HandleProvideOffer: No existing session with ID=%u.", sessionId);
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+            return;
+        }
+
+        // Verify that the session belongs to the accessing peer.
+        if (peerNodeId != existingSession->peerNodeID)
+        {
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+            return;
+        }
+
+        // Use the existing session for further processing.
+        outSession = *existingSession;
+    }
+
+    // Build and send the ProvideOfferResponse.
+    Commands::ProvideOfferResponse::Type resp;
+    resp.webRTCSessionID = outSession.id;
+    resp.videoStreamID.SetValue(outSession.videoStreamID);
+    resp.audioStreamID.SetValue(outSession.audioStreamID);
```

Comment:
Spec violation.  I said that last time; the comment was resolved, but not addressed, which is ... undesirable.

Reply:
Updated:

Case 1: Request HAS VideoStreamID
Request: { VideoStreamID: null } (auto-select)

Valid Response: { VideoStreamID: 42 } (provider-selected ID)
→ Field MUST appear because it was present in the request.

Case 2: Request OMITS VideoStreamID
Request: {} (no video)

Valid Response: No VideoStreamID field
→ Field MUST NOT appear, even if the provider later uses a video stream.